pkg/ipfs/node.go

```diff
@@ -53,6 +55,17 @@ func (c *IPFSConfig) NewCommunityIPFSNode(name string, path string) (error, stri
 		return err, "", "", s
 	}
 
+	// json struct of config : here we can modify it and write back
+	// ignore the peers for now (connect after bootstrapping?)
+	addr := data.Addresses.API
+	addrstring := addr[0]
+	parts := strings.Split(addrstring, "/")
```

Comment:
why are we doing this randomly?

Reply:
ah yeah this is confusing - i added a comment. This is for choosing a random port that ipfs config can use in lieu of some sort of port manager. Otherwise if you try to create more than one ipfs instance, they each default to using port 5001 and then conflict.

Reply:
sounds good - we should probably bite the bullet and do the port manager sooner rather than later